app/src/main/java/io/islnd/android/islnd/messaging/ServerTime.java

```diff
@@ -0,0 +1,48 @@
+package io.islnd.android.islnd.messaging;
+
+import android.content.Context;
+import android.util.Log;
+
+import java.io.IOException;
+
+public class ServerTime {
+    private static final String TAG = ServerTime.class.getSimpleName();
+    private static final int REPETITIONS = 2;
+
+    private static long serverOffsetMillis = 0;
+    private static boolean inSync = false;
+    public static boolean isInSync() {
+        return inSync;
+    }
+
+    public static long synchronize(Context context) throws IOException {
+        serverOffsetMillis = MessageLayer.getServerTimeOffsetMillis(context, REPETITIONS);
+        inSync = true;
+        return serverOffsetMillis;
+    }
+
+    public static void synchronizeManually(long newServerOffsetMillis) {
```

Comment:
This method seems to violate encapsulation. If we have a ServerTime class, and it has a `synchronize` method, I don't think we need this method. This is related to an idea I have about the shared preferences.